unionfs/unionfs.go

```diff
@@ -712,7 +712,12 @@ func (fs *unionFS) GetXAttr(name string, attr string, context *fuse.Context) ([]
 	}
 	r := fs.getBranch(name)
 	if r.branch >= 0 {
-		return fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		value, code := fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		// We intercept ENOSYS and replace it with ENODATA; returning ENOSYS would cause FUSE to never to GetXAttr() again.
+		if code == fuse.ENOSYS {
+			return nil, fuse.ENODATA
+		}
+		return value, code
```

Comment:
I'd note that that seems like the wrong methodology and would potentially create surprises for users, but fine. Shall I revert the incomplete work in https://github.com/hanwen/go-fuse/commit/7b281480b21a59f54a0c86451e7de13cd9be736f and change those ENODATA returns to ENOSYS?